modin/config/envvars.py

```diff
@@ -167,14 +167,28 @@ def put_if_default(cls, value):
         This method is used to set NPartitions from cluster resources internally
         and should not be called by a user.
         """
-        if cls._is_default:
+        if cls._is_default and not cls._is_set_from_config():
             cls.put(value)
 
     @classmethod
     def _get_default(cls):
         cls._is_default = True
         return CpuCount.get()
 
+    @classmethod
+    def _is_set_from_config(cls):
```

Comment:
It might be useful to be migrated to `Parameter` and made an internal flag for all configuration parameters... that way you can set it in `Parameter.get()` method (and it can be a tri-state - "set by config source", "set by default value", "set by user")

Right now you're duplicating the `.get()` efforts to an extent, which isn't necessary IMHO.

Reply:
In that case we will need to call `.get()` before `._put()` to initialize value of that flag that is not main goal of `.get()`. It is confusing a little, I think.

Reply:
On the other hand, we can leave a comment that it is a dummy call to initialize the flag. Thereby, we can get rid of code duplication that you mentioned.

Reply:
Changed way to set NPartitions from engines.